src/optimise.jl

```diff
@@ -34,17 +53,20 @@ function minimise(f, X::T, tol=1e-3) where {T}
         end
 
         # Remove all boxes whose lower bound is greater than the current one:
-        # Since they are ordered, just find the first one that is too big
-
-        cutoff = searchsortedfirst(working.data, (X, global_min), by=x->x[2])
-        resize!(working, cutoff-1)
+        
+        if structure == HeapedArray
```

Comment:
Replace this `if` by multiple dispatch by defining `delete_elements!` for both `Heap` and `SortedVectors`